runtime/gc_glue_java/CollectorLanguageInterfaceImpl.cpp

```diff
@@ -678,7 +678,8 @@ MM_CollectorLanguageInterfaceImpl::scavenger_switchConcurrentForThread(MM_Enviro
 			j9tty_printf(PORTLIB, "%p: Nursery [%p,%p] Evacuate [%p,%p] GS [%p,%p] Section size 0x%zx, sections %lu bit offset %lu bit mask 0x%zx\n",
 					vmThread, nurseryBase, nurseryTop, gsBase, gsTop, pageBase, pageTop, _extensions->getConcurrentScavengerPageSectionSize() , sectionCount, startOffsetInBits, bitMask);
 		}
-
+		vmThread->evacuateBase = gsBase;
+		vmThread->evacuateTop = gsTop;
 		vmThread->privateFlags |= J9_PRIVATE_FLAGS_CONCURRENT_SCAVENGER_ACTIVE;
 		j9gs_enable(&vmThread->gsParameters, _extensions->getConcurrentScavengerPageStartAddress(), _extensions->getConcurrentScavengerPageSectionSize(), bitMask);
     } else {
```

Comment:
For the 'else' clause, I was expecting to see something like:

evacBase  = UINT_MAX; (or heapTop)
evacTop = 0;  (or heapBase)

effectively making any address out of range (and failing it on just one of the checks), so that you don't need to explicitly check if-scavenge-in-progress flag.